### DISCUSSION TITLE
Avoiding com.google.common in the birt-runtime

### DISCUSSION THREAD

**Discussion by Miurath** [eclipse-birt/birt#1217] [Q&A]
I am currently updating our BIRT from version 4.4.2 to 4.12.

When I tried to update the birt-runtime library, I noticed that the birt-runtime now contains some of the com.google.common libraries. 
We already use those libraries, but with different versions in our software and having both of them in the buildpath causes problems.

I can think of these two options to fix this:

Delete the problematic libraries from the birt-runtime.

Or:

Find out how the birt-runtime is build and use only the relevant jar's and not the whole birt-runtime-jar.

How should I deal with this?

**Comment by hvbtup** [comment]
I'm not sure, but these Google libs (which exactly do you mean?) are probably used only for the Apache Cassandra datasource, so what I would do is just remove them.

**Comment by Miurath** [comment]
I am referring to everything in org.eclipse.birt.runtime_4.12.0-20221128.jar\com\google\common.
Those libraries were not there in 4.4.2.

So you would suggest just deleting those libraries from the jar?

**Reply by hvbtup** [reply]
Yes, that's would I would try. Of course you should check if BIRT and your reports still work then.
